src/lib/contracts.ts

```diff
@@ -198,67 +255,37 @@ export function validateContract(contract: unknown): boolean {
 
 	return true;
 }
+
 export function validateTargetContracts(
 	apps: TargetApps,
-): Dictionary<ApplicationContractResult> {
-	return Object.keys(apps)
-		.map((appUuid): [string, ApplicationContractResult] => {
-			const app = apps[appUuid];
-			const [release] = Object.values(app.releases);
-			const serviceContracts = Object.keys(release?.services ?? [])
-				.map((serviceName) => {
-					const service = release.services[serviceName];
-					const { contract } = service;
-					if (contract) {
-						try {
-							// Validate the contract syntax
-							validateContract(contract);
-
-							return {
-								serviceName,
-								contract,
-								optional: checkTruthy(
-									service.labels?.['io.balena.features.optional'],
-								),
-							};
-						} catch (e: any) {
-							throw new ContractValidationError(serviceName, e.message);
+): Dictionary<AppContractResult> {
+	return Object.fromEntries(
```

Comment:
I would probably avoid the `fromEntries` and instead do
```ts
const result = {};
for (const [appUuid, app] of Object.entries(apps)) {
	...
	result[appUuid] = appContractResult;
}
return result;
```

It will reduce the nesting a bit, make it clearer that the end goal is to populate a `{ [appUuid: string]: ContractResult }` type object, and also have a marginal benefit in reduced allocations (it won't allocate an array of arrays to then convert to an object)